hummingbot/connector/gateway/clob_spot/gateway_clob_spot.py

```diff
@@ -121,7 +124,10 @@ def trading_pairs(self) -> List[str]:
 
     @property
     def is_cancel_request_in_exchange_synchronous(self) -> bool:
-        return False
+        if hasattr(self._api_data_source, 'is_cancel_request_in_exchange_synchronous'):
```

Comment:
Yeah, we really didn't want to touch this one here but the problem is the data source base classes (GatewayCLOBAPIDataSourceBase or CLOBAPIDataSourceBase) doesn't have this method available there and then we cannot call it. We used this technique that we've found somewhere in hummingbot before to enable the possibility of the method to be called if it exists.

But if it is ok for you, we could touch some more files and we could replicate this method on the CLOBAPIDataSourceBase and make it return False by default and then we could override it there for us (in our case returning True).

Because GatewayCLOBSpot and CLOBAPIDataSourceBase have different hierarchies some of the functionalities are not available in one another.

Reply:
and how is possible that the cancel request is synchronous? don't you have to wait for a tx confirmation?

Reply:
We actually wait for the tx confirmation (the operation is async in term that we need to use an "await" to wait for it).

But in this case (at least from what we've understood), the GatewayCLOBSpot class implementation concerns more if the cancellation request was dispatched and later checked or if it is expected it to return the tx by the end of the execution of the method (which is our case).